openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ServiceToOpenAPIConverterUtils.java

```diff
@@ -159,38 +152,53 @@ private static void extractListenersAndServiceNodes(String serviceName, List<Str
-                            .forEachOrdered(listenerType -> extractServiceDeclarationNodes(serviceName,
-                                    availableService, servicesToGenerate, serviceDeclarationNode));
-                }
-            }
-        }
-    }
+                if (isHttpService(serviceNode, semanticModel)) {
+                    Optional<Symbol> serviceSymbol = semanticModel.symbol(serviceNode);
+                    if (serviceSymbol.isPresent() && serviceSymbol.get() instanceof ServiceDeclarationSymbol) {
+                        String service = OpenAPIEndpointMapper.ENDPOINT_MAPPER.getServiceBasePath(serviceNode);
+                        /*
+                          `String updateServiceName` used to track the service name for service file contains
+                          multiple service node.
+                          ex;
+                          <pre>
+                              listener http:Listener ep1 = new (443, config = {host: "pets-tore.swagger.io"});
+                              service /hello on ep1 {
+                                  resource function post hi(@http:Payload json payload) {
+                                 }
+                              }
+                              service /hello on new http:Listener(9090) {
+                                  resource function get hi() {
+                                  }
+                              }
+                          </pre>
+                          Using absolute path we generate file name, therefore having same name may overwrite
+                          the file, due to this suppose to use hashcode as identity factor for the file name.
 
-    /**
-     * Filter all the serviceNodes in syntax tree.
-     */
-    private static void extractServiceDeclarationNodes(String serviceName, List<String> availableService,
-                                                       List<ServiceDeclarationNode> servicesToGenerate,
-                                                       ServiceDeclarationNode serviceNode) {
-        if (serviceName != null) {
-            // Filtering by service name
-            String service = OpenAPIEndpointMapper.ENDPOINT_MAPPER.getServiceBasePath(serviceNode);
-            availableService.add(service);
-            if (serviceName.equals(service)) {
-                servicesToGenerate.add(serviceNode);
+                          Generated file name for above example -> hello_openapi.yaml, hello_45673_openapi
+                          .yaml
+                         */
+                        String updateServiceName = service;
+                        if (servicesToGenerate.containsKey(service)) {
+                            updateServiceName = service + HYPHEN + serviceSymbol.get().hashCode();
+                        }
+                        if (serviceName != null) {
+                            // Filtering by service name
+                            availableService.add(service);
```

Comment:
WDYM by `availableService`? It seems like a list of resolved service nodes to me.

Reply:
This is used to store all the http services absolute paths contain in the ballerina file. This list maintains in case some user has provided some specific absolute path that does not exist in the service file for generating contract, then that time we can generate error message by adding all the service name exist in service by using this list details. :)